File: BinaryTreeInorderTra.go

 package main
 // Definition for a binary tree node.
type TreeNode struct {
      Val int
      Left *TreeNode
      Right *TreeNode
  }
 
 func inorderTraversal(root *TreeNode) (res []int) {
	if root == nil {
		return 
	}
	
	var traversal func(node *TreeNode) 
	traversal = func(node *TreeNode) {
		if node == nil {
			return
		}
		traversal(node.Left)
		res = append(res, node.Val)
		traversal(node.Right)
	}
	traversal(root)
	return res
}

File: BinaryTreeInorderTra.py
# Definition for a binary tree node.

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            #确定递归终止条件
            return []
        
        # 中序遍历： 中左右顺序
        # 确定单层递归逻辑及传入参数
        left = self.inorderTraversal(root.left)
        right = self.inorderTraversal(root.right)

        return left + [root.val] + right